pallets/withdraw-teleport/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+			(*beneficiary).try_into().map_err(|()| pallet_xcm::Error::<T>::BadVersion)?;
+		//Unbox fee asset
+		let fee_asset: MultiAssets =
+			(*fee_asset).try_into().map_err(|()| pallet_xcm::Error::<T>::BadVersion)?;
+
+		//Create assets
+
+		// Native from local perspective
+		let native_asset = MultiAsset {
+			id: AssetId::Concrete(MultiLocation::here()),
+			fun: Fungibility::Fungible(native_asset_amount),
+		};
+		let assets = MultiAssets::from(vec![native_asset.clone()]);
+
+		// Native from foreign perspective
+		let context = T::UniversalLocation::get();
+		let native_as_foreign = native_asset
+			.reanchored(&dest, context)
+			.map_err(|_| pallet_xcm::Error::<T>::CannotReanchor)?;
+		let foreing_assets = MultiAssets::from(vec![native_as_foreign]);
+
+		// TeleportFilter check
+		let value = (origin_location, assets.into_inner());
+		ensure!(T::XcmTeleportFilter::contains(&value), pallet_xcm::Error::<T>::Filtered);
+		let (origin_location, assets) = value;
+
+		// Reanchor the fee asset to the destination chain.
+		let fee_asset_item: usize = 0;
+		let fees = fee_asset
+			.get(fee_asset_item as usize)
+			.ok_or(pallet_xcm::Error::<T>::Empty)?
+			.clone()
+			.reanchored(&dest, context)
+			.map_err(|_| pallet_xcm::Error::<T>::CannotReanchor)?;
+
+		// TODO: Define if Withdrawn fee assets are deposited or trapped.
+		// Check if there is no vulnerability through RefundSurplus
+		// let max_assets = (assets.len() as u32).checked_add(1).ok_or(Error::<T>::TooManyAssets)?;
+
+		// DISCLAIMER: Splitting the instructions to be executed on origin and destination is
+		// discouraged. Due to current limitations, we need to generate a message
+		// to be executed on origin and another message to be sent to be executed on destination in
+		// two different steps as:
+		// - We cannot buy execution on Asset Hub with foreign assets.
+		// - We cannot send arbitrary instructions from a local XCM execution.
+		// - InitiateTeleport prepends unwanted instructions to the message.
+		// - Asset Hub does not recognize Sibling chains as trusted teleporters of ROC.
+
+		//Build the message to execute on origin.
+		let assets: MultiAssets = assets.into();
+		let message: Xcm<<T as frame_system::Config>::RuntimeCall> =
+			Xcm(vec![WithdrawAsset(assets.clone()), BurnAsset(assets)]);
+
+		// Build the message to send to be executed.
+		// Set WeightLimit
+		// TODO: Implement weight_limit calculation with final instructions.
+		let weight_limit: WeightLimit = Unlimited;
+		let xcm_to_send: Xcm<()> = Xcm(vec![
+			// There are currently no limitations on the amount of fee assets to withdraw.
+			// Since funds are withdrawn from the Sovereign Account of the origin, chains must be
+			// aware of this and implement a mechanism to prevent draining.
+			WithdrawAsset(fee_asset),
+			BuyExecution { fees, weight_limit },
+			ReceiveTeleportedAsset(foreing_assets.clone()),
+			// Intentionally trap ROC to avoid exploit of draining Sovereing Account
```

Comment:
Doesn't look trapped?

Reply:
I would need some more explanation here as I do not fully understand where are you suggesting to debit the ROC from. Thanks!

Reply:
I mean here on Trappist. You have reserve-backed ROC here, right?

Reply:
No initially, but yes after #275 , so shall we move forward and merge 275 which solves this ?